spec/services/email_alert_api/subscriber_list_updater_spec.rb

```diff
@@ -0,0 +1,172 @@
+        before do
+          stub_email_alert_api_creates_subscriber_list(
+            "links" => { "document_collections" => [SecureRandom.uuid] },
+            "content_id" => successor_content_id,
+            "slug" => "document_collection_slug",
+          )
+        end
+
+        it "successor has a mapped specialist topic content id that matches the specialist topic's content id" do
+          data = base_content_item_data
+          data["details"] = { "mapped_specialist_topic_content_id" => topic.content_id }
+          successor = ContentItem.new(data)
+
+          bulk_migrate_stub = stub_request(:post, bulk_migrate_link)
+                                .with(body: { to_slug: "document_collection_slug", from_slug: "tax-credits-and-child-benefit-child-benefit" })
+                                .to_return(status: 200)
+
+          expect(Services.email_alert_api).to receive(:bulk_unsubscribe).never
+
+          described_class.call(item: topic, successor:)
+
+          expect(bulk_migrate_stub).to have_been_requested
+        end
+      end
+
+      context "moving subscribers to a taxonomy topic subscriber list" do
+        let(:taxonomy_topic_content_id) { "b20215a9-25fb-4fa6-80a3-42e23f5352c2" }
+        let(:taxonomy_topic_base_path) { "/money/dealing-with-hmrc" }
+        let(:links_data_for_taxonomy_topic) do
+          [
+            {
+              "content_id" => taxonomy_topic_content_id,
+              "base_path" => taxonomy_topic_base_path,
+              "document_type" => "taxon",
+              "title" => "Dealing with HMRC",
+            },
+          ]
+        end
+
+        before do
+          stub_email_alert_api_creates_subscriber_list(
+            "links" => { "taxon_tree" => [taxonomy_topic_content_id] },
+            "title" => "Dealing with HMRC",
+            "url" => taxonomy_topic_base_path,
+            "slug" => "taxonomy_topic_slug",
+          )
+        end
+
+        it "successor has a taxonomy topic email override only" do
+          data = base_content_item_data
+          base_content_item_data["links"] = { "taxonomy_topic_email_override" => links_data_for_taxonomy_topic }
+          successor = ContentItem.new(data)
+
+          stub_content_store_has_item(taxonomy_topic_base_path)
+          bulk_migrate_to_taxonomy_topic_stub = stub_request(:post, bulk_migrate_link)
+                                .with(body: { to_slug: "taxonomy_topic_slug", from_slug: "tax-credits-and-child-benefit-child-benefit" })
+                                .to_return(status: 200)
+
+          expect(Services.email_alert_api).to receive(:bulk_unsubscribe).never
+
+          described_class.call(item: topic, successor:)
+
+          expect(bulk_migrate_to_taxonomy_topic_stub).to have_been_requested
+        end
+
+        it "successor has a mapped specialist topic content id that matches the specialist topic's content id and also a taxonomy topic email override" do
```

Comment:
Am I right in stating that the `it` is describing the setup here (and not the behavioural expectations)?  e.g. should it actually be `it "should subscribe to associated taxonomy topic"?

Reply:
You're right, and this is a consequence of the first commit where I've grouped the context blocks by the outcome: migrate or unsubscribe. Which is unconventional, and I obsessed over this file for a long time 😂 But I think (not sure if you'll agree!) that if you read the whole file overall it makes the behaviour of the service easier to understand.